src/apis/token_api.ts

```diff
@@ -1,18 +1,32 @@
 import * as Web3 from "web3";
+import * as singleLineString from "single-line-string";
+
 import { Web3Utils } from "../../utils/web3_utils";
 import { ContractsAPI } from "./";
 import { BigNumber } from "bignumber.js";
+import { Assertions } from "../invariants";
 import { TxData } from "../types";
 
 const TRANSFER_GAS_MAXIMUM = 70000;
 
+export const TokenAPIErrors = {
+    INSUFFICIENT_SENDER_BALANCE: () =>
```

Comment:
Let's add an argument to this that specifies the spender's address -- i.e. `"Sender with address ${spenderAddress} does not have sufficient balance..."`